app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,56 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!, only: [:new, :create, :edit, :update, :destroy]
+
+  # index
+  def index
+    @post = Post.find(params[:post_id])
+    @comments = @post.comments.all
+  end
+
+  # new
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new
+  end
+
+  # create
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.create(comment_params)
+
+    redirect_to post_path(@post)
```

Comment:
For some reason, in the deployed version, this is redirecting to `https://scribbleaf.herokuapp.com/posts/1/comments.1` or whatever the relevant number is...

Check your `rake routes` and maybe inspect some objects in `rails c` to see if everything looks normal...might be that your associations aren't working the way you expect them to...